### PR TITLE
[12.0][MIG] mrp_bom_version: Migration to 12.0

### PR DESCRIPTION
**Pull request by vvrossem**
- [Original module in 8.0](https://github.com/OCA/manufacture/tree/8.0/mrp_bom_version)
- [Applied changes done in #105 for 9.0](https://github.com/OCA/manufacture/pull/105)
- Followed OCA's guidelines for migration 10.0 to 12.0
- Tested only on 12.0

### CODE REVIEW

File: mrp_bom_version/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+{
+    "name": "MRP - BoM version",
+    "summary": "BoM versioning",
+    "version": "12.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "OdooMRP team,"
+    "AvanzOSC,"
+    "Serv. Tecnol. Avanzados - Pedro M. Baeza, "
+    "Odoo Community Association (OCA)",
+    "website": "http://www.odoomrp.com",
```
Comment (houssine78) [inline mrp_bom_version/__manifest__.py]: the site doesn't seem to exist anymore could put the link to the oca repo instead ?

Comment (vvrossem) [reply mrp_bom_version/__manifest__.py]: done :)

File: mrp_bom_version/models/mrp_bom.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    product_tmpl_id = fields.Many2one(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    product_id = fields.Many2one(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    product_qty = fields.Float(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    code = fields.Char(states={"historical": [("readonly", True)]})
```
Comment (houssine78) [inline mrp_bom_version/models/mrp_bom.py:72]: it's cosmetic but fields description aren't all formatted the same way.

Comment (pedrobaeza) [reply mrp_bom_version/models/mrp_bom.py:72]: `black` formats them according if they fit into one line or not given the line length limit.

**Review by houssine78 — APPROVED** [review]

**Pull request closed without merging**